src/core/Effect.cpp

```diff
@@ -181,26 +169,56 @@ Effect * Effect::instantiate( const QString& pluginName,
 
 
 
-void Effect::checkGate(double outSum)
+void Effect::handleAutoQuit(std::span<const SampleFrame> output)
 {
-	if( m_autoQuitDisabled )
+	if (m_autoQuitDisabled)
 	{
 		return;
 	}
 
-	// Check whether we need to continue processing input.  Restart the
+	/*
+	 * In the past, the RMS was calculated then compared with a threshold of 10^(-10).
+	 * Now we use a different algorithm to determine whether a buffer is non-quiet, so
+	 * a new threshold is needed for the best compatibility. The following is how it's derived.
+	 *
+	 * Old method:
+	 * RMS = average (L^2 + R^2) across stereo buffer.
+	 * RMS threshold = 10^(-10)
+	 *
+	 * So for a single channel, it would be:
+	 * RMS/2 = average M^2 across single channel buffer.
```

Comment:
If `M` is for a single channel, I think `C` (for Center) might work better. I guess the central problem though is that developers need to know how to read things like this that require knowing certain terminology.